global/scripts/mixtureModel.sh

```diff
@@ -0,0 +1,139 @@
+  local outFile0="$outFile"
+  local tDir
+  tDir=$(mktemp -d)
+  local FSLOUTPUTTYPE0="$FSLOUTPUTTYPE"
+  local stat=0
+  local ext
+
+  # convert input if needed
+  if [[ "$inFile0" == *dscalar.nii ]]; then
+    # convert input cifti to nifti
+    inFile="$(mktemp --suffix=.nii)"
+    $wbcmd -cifti-convert -to-nifti "$inFile0" "$inFile" -smaller-dims
+    # local dims
+    dims=$($wbcmd -file-information "$inFile" | grep Dimensions | awk -F': ' '{print $2}')
+    # shellcheck disable=SC2206
+    dims_arr=($dims)
+    for i in "${!dims_arr[@]}"; do
+      if [[ "${dims_arr[$i]}" == "1" && $i -lt 3 ]]; then
+        echo "warning: singleton dimension in converted nifti, melodic may not interpret correctly!" >&2
+        break
+      fi
+    done
+    inFile="${inFile%.nii}"
+  elif [[ "$inFile0" == *.nii ]]; then
+    inFile="${inFile0%.nii}"
+  elif [[ "$inFile0" == *.nii.gz ]]; then
+    inFile="${inFile0%.nii.gz}"
+  else
+    echo "inFile is not a nifti or dscalar cifti?" >&2
+    return 1
+  fi
+
+  # handle output extension
+  if [[ "$outFile0" == *dscalar.nii ]]; then
+    if [[ "$inFile0" != *dscalar.nii ]]; then
+      echo "cifti output only supported for cifti input." >&2
+      return 1
+    fi
+    outFile="${outFile0%.dscalar.nii}"
+    FSLOUTPUTTYPE="NIFTI_GZ"
+    ext="nii.gz"
+  elif [[ "$outFile0" == *.nii ]]; then
+    outFile="${outFile0%.nii}"
+    FSLOUTPUTTYPE="NIFTI"
+    ext=".nii"
+  elif [[ "$outFile0" == *.nii.gz ]]; then
+    FSLOUTPUTTYPE="NIFTI_GZ"
+    ext="nii.gz"
+    outFile="${outFile0%.nii.gz}"
+  else
+    echo "outFile is not a nifti or dscalar cifti?" >&2
+    return 1
+  fi
+
+  # run gaussian mixture modeling with melodic
+  mkdir -p "$tDir"
+  echo "1" > "$tDir/grot"
+  $melocmd -i "$inFile" --ICs="$inFile" --mix="$tDir/grot" -o "$tDir" --Oall --report -v --mmthresh=0
+
+  # check for multiple z-score maps
+  mapCounts=$(find "$tDir/stats/" -name 'thresh_zstat*' -exec bash -c 'fslval "$1" dim4' _ {} \;)
+  maxCount=0
+  for count in $mapCounts; do
+    if (( count > maxCount )); then maxCount=$count; fi
+  done
+  if (( maxCount > 1 )); then
+    echo "warning: At least one component returned two z-score maps (alpha = 0.05 and 0.01)! Using first map only." >&2
+  fi
+
+  # concatenate volumes
+  $wbshrtctscmd -volume-concatenate -map 1 "${outFile}.${ext}" $(ls "$tDir"/stats/thresh_zstat* | sort -V)
+
+  # clean up temporary files
+  rm -r "$tDir"
+
+  # convert output to cifti, if needed
+  if [[ "$outFile0" == *dscalar.nii ]]; then
+    $wbcmd -cifti-convert -from-nifti "${outFile}.nii.gz" "$inFile0" "$outFile0"
+    imrm "$inFile" "$outFile"
+  elif [[ "$inFile0" == *dscalar.nii ]]; then
+    imrm "$inFile"
+  fi
+
+  # restore FSLOUTPUTTYPE if needed
+  export FSLOUTPUTTYPE="$FSLOUTPUTTYPE0"
+}
+# Call the function if the script is run directly
+if [[ "${BASH_SOURCE[0]}" == "$0" ]]; then
+  mixtureModel "$@"
+fi
+# Usage example:
+# mixtureModel input.dscalar.nii output.dscalar.nii
```

Comment:
This is pythonish stuff to guard against an executable script being sourced (included) to use it as a function.  We don't need this, we separate our bash into scripts and libraries (and scripts can simply run other scripts, without being weird about it).  The implementation doesn't need to be a function at all.